### PR TITLE
Return eos_token from decoding algos

### PR DESCRIPTION
**Pull request by as-suvorov**

### CODE REVIEW

**Comment by as-suvorov** [comment]
Waiting for https://github.com/openvinotoolkit/openvino_tokenizers/pull/160

**Comment by ilya-lavrenov** [comment]
You can migrate to latest 2024.2 OpenVINO Tokenizers branch.

File: src/cpp/src/text_callback_streamer.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@ TextCallbackStreamer::TextCallbackStreamer(const Tokenizer& tokenizer, std::func
 bool TextCallbackStreamer::put(int64_t token) {
     std::stringstream res;
     // do nothing if <eos> token is met and if print_eos_token=false
+    // todo: remove when skip_special_tokens = True applied for tokenizer by default
```
Comment (ilya-lavrenov) [inline src/cpp/src/text_callback_streamer.cpp]: drop?

Comment (as-suvorov) [reply src/cpp/src/text_callback_streamer.cpp]: sure, done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**